packages/protocol/contracts/common/GoldToken.sol

```diff
@@ -34,12 +37,29 @@ contract GoldToken is
   // Burn address is 0xdEaD because truffle is having buggy behaviour with the zero address
   address constant BURN_ADDRESS = address(0x000000000000000000000000000000000000dEaD);
 
+  IMintGoldSchedule public goldTokenMintingSchedule;
+
   event Transfer(address indexed from, address indexed to, uint256 value);
 
   event TransferComment(string comment);
 
   event Approval(address indexed owner, address indexed spender, uint256 value);
 
+  event SetGoldTokenMintingScheduleAddress(address indexed newScheduleAddress);
+
+  modifier isRestricted() {
+    if (!isL2()) {
+      require(msg.sender == address(0), "Only VM can call");
+    } else {
+      require(
+        (msg.sender == address(goldTokenMintingSchedule) || msg.sender == owner()) ||
+          msg.sender != address(0),
```

Comment:
|| msg.sender != address(0)

seems to allow everyone to mint - shouldn't it be && ?

Do we need this check at all ? Only owner() or goldTokenMintingSchedule should be allowed

Reply:
good catch. Fixed.